src/components/page/page_header/page_header_content.tsx

```diff
@@ -344,25 +344,22 @@ export const EuiPageHeaderContent: FunctionComponent<EuiPageHeaderContentProps>
 
   let rightSideFlexItem;
   if (rightSideItems && rightSideItems.length) {
-    const wrapWithFlex = () => {
-      return rightSideItems.map((item, index) => {
-        return (
-          <EuiFlexItem grow={false} key={index}>
-            {item}
-          </EuiFlexItem>
-        );
-      });
+    const wrapWithFlex = (reverse: boolean) => {
+      const itemsToRender = reverse
+        ? [...rightSideItems].reverse()
+        : rightSideItems;
+
+      return itemsToRender.map((item, index) => (
+        <EuiFlexItem grow={false} key={index}>
+          {item}
+        </EuiFlexItem>
+      ));
     };
```

Comment:
code style nit: TBH, I don't think we need a function here at all as opposed to just defining a React JSX var. (FWIW, I'm aware Caroline originally wrote with that syntax and we were just following her existing structure, but I would like to clean it up while we're here as I think the passed arg is harder to read than writing it out statically).

```suggestion
    const itemsToRender = isResponsiveBreakpoint
      ? rightSideItems
      : [...rightSideItems].reverse();

    const rightSideFlexItems = itemsToRender.map((item, index) => (
      <EuiFlexItem grow={false} key={index}>
        {item}
      </EuiFlexItem>
    ));
```